toolkit/utils/ogmios-client/src/jsonrpsee.rs

```diff
@@ -2,35 +2,41 @@
 //! Major drawback is that it swallows the error response from the server in case of 400 Bad Request.
 
 use crate::{OgmiosClient, OgmiosClientError, OgmiosParams};
-use jsonrpsee::core::params::ArrayParams;
 use jsonrpsee::{
-	core::{client::ClientT, params::ObjectParams},
+	core::{client::ClientT, traits::ToRpcParams, ClientError},
 	http_client::HttpClient,
 };
 use serde::de::DeserializeOwned;
 
+fn request_to_json(req: impl ToRpcParams) -> Result<String, OgmiosClientError> {
+	let json_str = match req.to_rpc_params().expect("Parameters are correct") {
```

Comment:
Please map error with `serde_error_to_parameters_error` or change return type, because with .expect id doesn't return Err.

Also, there is no need to match, `to_rpc_params().map_err(...).unwrap_or_default()` will yield `Box<RawValue>` that has display.